ring/model.go

```diff
@@ -302,13 +303,17 @@ func normalizeIngestersMap(inputRing *Desc) {
 	}
 }
 
+var tokenMapPool = sync.Pool{New: func() interface{}{return make(map[uint32]bool)}}
+
 func conflictingTokensExist(normalizedIngesters map[string]InstanceDesc) bool {
 	count := 0
 	for _, ing := range normalizedIngesters {
 		count += len(ing.Tokens)
 	}
-
-	tokensMap := make(map[uint32]bool, count)
+	tokensMap := tokenMapPool.Get().(map[uint32]bool)
+	defer func() {
+		tokenMapPool.Put(tokensMap)
```

Comment:
You either need to clear the map after taking it out or before putting it back. You're getting previous state :(

Reply:
Yeah, too bad there's not a good way to .clear() the map :/

Reply:
(Problem here being over time, eventually all tokens will be in the map, and the check will say there are always conflicting tokens, so negating the original optimization)

Reply:
do we still need the `count`? it was only used to size the map I think.

Reply:
I went with the delete here to clear this map, but also we could just use a map[uint32]{} here instead of using a boolean value and check contains rather than the value, but it's just a space optimization.

Reply:
Have we tried to create empty map with zero size instead?

Reply:
We don't use count here, good point. I tried a make(...) and that was slower than clearing the map oddly, but maybe I was looking at it wrong.

Reply:
we can also try using `map[uint32]struct{}` which should be smaller, but requires bit of extra code to use.

Reply:
looks like I should reload the page before adding more comments :)

Reply:
I tested again with the zero size make() and it was way worse on cpu and memory than deleting the keys, which is an interesting go question.

Reply:
Thanks. Not sure if it is worth trying to find conflicting tokens without any map at all which should be easy since all tokens slices are sorted already.

Reply:
[nit] After think about this, I think clearing the map after `Get()` is preferable, because then you're not trusting the last user to do the clearing, but this is a nit.

Reply:
We can leave this as-is